src/RarityScore.ts

```diff
@@ -236,22 +242,32 @@ export function toEpoch(timestamp: BigInt): i32 {
   let ts = timestamp.toI32()
   let firstEpochTs = OTTOPIA_RARITY_SCORE_RANKING_FIRST_EPOCH
   let duration = OTTOPIA_RARITY_SCORE_RANKING_DURATION
+  let epoch3extendTs = 86400 * 2
   // log.warning('toEpoch ts {}, firstEpochTs {}, duration {}', [
   //   ts.toString(),
   //   firstEpochTs.toString(),
   //   duration.toString(),
   // ])
   if (ts < firstEpochTs) {
     return 0
+  } else if (ts >= firstEpochTs && ts < firstEpochTs + 3 * duration) {
+    return (ts - firstEpochTs) / duration
+  } else if (ts >= firstEpochTs + 3 * duration && ts < firstEpochTs + epoch3extendTs + 4 * duration) {
+    return 3
+  } else {
+    return (ts + epoch3extendTs - firstEpochTs) / duration
   }
-  // log.warning('toEpoch (ts - firstEpochTs) / duration {}', [((ts - firstEpochTs) / duration).toString()])
-  return (ts - firstEpochTs) / duration
 }
 
 function toEpochEndTimestamp(epoch: i32): BigInt {
   let firstEpochTs = OTTOPIA_RARITY_SCORE_RANKING_FIRST_EPOCH
   let duration = OTTOPIA_RARITY_SCORE_RANKING_DURATION
-  return BigInt.fromI64(firstEpochTs + duration * (epoch + 1))
+  let epoch3extendTs = 86400 * 2
```

Comment:
Share this value with L245

Reply:
https://github.com/OtterClam/otto-subgraph/pull/21/commits/469b3a5ae64aae407c04fa569e40ca0c50180c89

Reply:
https://github.com/OtterClam/otto-subgraph/pull/21/commits/529763341d05441cb2d397c4ac417f95a1b2d811